public/js/pimcore/object/classes/data/multiselect.js

```diff
@@ -253,84 +391,141 @@ pimcore.object.classes.data.multiselect = Class.create(pimcore.object.classes.da
                 })
             ]
         });
-
-        this.selectionModel = valueGrid.getSelectionModel();
-
-        let appendedStylingItems = stylingItems.concat([
-            {
-                xtype: "numberfield",
-                fieldLabel: t("maximum_items"),
-                name: "maxItems",
-                value: datax.maxItems,
-                minValue: 0
-            },
-            {
-                xtype: "combo",
-                fieldLabel: t("multiselect_render_type"),
-                name: "renderType",
-                itemId: "renderType",
-                mode: 'local',
-                store: [
-                    ['list', 'List'],
-                    ['tags', 'Tags']
-                ],
-                value: datax["renderType"] ? datax["renderType"] : 'list',
-                triggerAction: "all",
-                editable: false,
-                forceSelection: true
-            }
-        ]);
-
-        appendedStylingItems = appendedStylingItems.concat(
-            pimcore.object.helpers.selectField.getOptionsProviderFields(datax, valueGrid)
-        );
-        appendedStylingItems.push(valueGrid);
-
-        return appendedStylingItems;
     },
 
-    applyData: function ($super) {
-
-        $super();
+    createdefaultvaluegrid: function (store) {
+        return Ext.create('Ext.grid.Panel', {
+            itemId: "defaultvalueeditor",
+            viewConfig: {
+                plugins: [
+                    {
+                        ptype: 'gridviewdragdrop',
+                        dragroup: 'objectclassselect'
+                    }
+                ]
+            },
+            tbar: [{
+                xtype: "tbtext",
+                text: t("default_value")
+            }, "-", {
+                xtype: "button",
+                iconCls: "pimcore_icon_add",
+                handler: function () {
+                    let u = {
+                        value: ""
+                    };
+                    let selection = this.defaultValueModel.getSelection();
+                    let idx;
+                    let count = store.getCount();
 
-        var options = [];
+                    if (selection.length > 0) {
+                        let selectedRow = selection[0];
+                        idx = store.indexOf(selectedRow) + 1;
+                    } else {
+                        idx = count;
+                    }
 
-        var valueEditor = this.specificPanel.getComponent("valueeditor");
-        if (valueEditor) {
-            var valueStore = valueEditor.getStore();
-            valueStore.commitChanges();
-            valueStore.each(function (rec) {
-                options.push({
-                    key: rec.get("key"),
-                    value: rec.get("value")
-                });
-            });
-        }
+                    if (count >= this.datax.maxItems){
```

Comment:
```suggestion
                    if (this.datax.maxItems > 0 && count >= this.datax.maxItems){
```
otherwise it would be always reaching the limit